fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java

```diff
@@ -1695,16 +1700,201 @@ private void handleShowRestore() throws AnalysisException {
+                            grantPrivilegeStmt.setActionList(privilegeEntry.getActionSet());
+                            grantPrivilegeStmt.setObjectList(Lists.newArrayList(privilegeEntry.getObject()));
+                            info.add(AstToStringBuilder.toString(grantPrivilegeStmt));
+
+                            infos.add(info);
+                        }
+                    }
+
+                    resultSet = new ShowResultSet(showStmt.getMetaData(), infos);
+                }
+            } catch (PrivilegeException e) {
+                throw new SemanticException(e.getMessage());
+            }
+        } else {
+            List<List<String>> infos = GlobalStateMgr.getCurrentState().getAuth().getGrantsSQLs(showStmt.getUserIdent());
+            resultSet = new ShowResultSet(showStmt.getMetaData(), infos);
+        }
     }
 
-    private void handleShowRoles() {
+    private void handleShowRoles() throws AnalysisException {
         ShowRolesStmt showStmt = (ShowRolesStmt) stmt;
-        // TODO(yiming): handle show roles in new RBAC privilege framework
-        List<List<String>> infos = GlobalStateMgr.getCurrentState().getAuth().getRoleInfo();
-        resultSet = new ShowResultSet(showStmt.getMetaData(), infos);
+
+        if (GlobalStateMgr.getCurrentState().isUsingNewPrivilege()) {
+            try {
+                List<List<String>> infos = new ArrayList<>();
+                PrivilegeManager privilegeManager = GlobalStateMgr.getCurrentState().getPrivilegeManager();
+                Map<Long, RolePrivilegeCollection> rolePrivilegeCollectionMap = privilegeManager.getRoleIdToPrivilegeCollection();
+                for (Map.Entry<Long, RolePrivilegeCollection> rolePrivilegeCollectionEntry
+                        : rolePrivilegeCollectionMap.entrySet()) {
+                    List<String> info = new ArrayList<>();
+                    RolePrivilegeCollection rolePrivilegeCollection =
+                            privilegeManager.getRolePrivilegeCollectionUnlocked(rolePrivilegeCollectionEntry.getKey(), true);
+                    info.add(rolePrivilegeCollection.getName());
+                    infos.add(info);
+                }
+
+                resultSet = new ShowResultSet(showStmt.getMetaData(), infos);
+
+            } catch (PrivilegeException e) {
+                throw new AnalysisException(e.getMessage());
+            }
+        } else {
+            List<List<String>> infos = GlobalStateMgr.getCurrentState().getAuth().getRoleInfo();
+            resultSet = new ShowResultSet(showStmt.getMetaData(), infos);
+        }
+    }
+
+    private void handleShowUser() {
+        List<List<String>> rowSet = Lists.newArrayList();
+
+        ShowUserStmt showUserStmt = (ShowUserStmt) stmt;
+        if (showUserStmt.isAll()) {
+            Set<UserIdentity> userIdentities =
+                    GlobalStateMgr.getCurrentState().getPrivilegeManager().getUserToPrivilegeCollection().keySet();
```

Comment:
I think we should use AuthenticationManager#getUserToAuthenticationInfo() here, because user may not have any privilege in PrivilegeManager, and also read lock protection.